modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -708,50 +797,50 @@ public Table createTable(String name, Consumer<TableChange> tableInitChange) {
-                    );
-                }).whenComplete((res, t) -> {
-                    if (t != null) {
-                        Throwable ex = getRootCause(t);
+                        tableInitChange.accept(ch);
+
+                        var extConfCh = ((ExtendedTableChange) ch);
+
+                        tableCreateFuts.put(extConfCh.id(), tblFut);
+
+                        // Affinity assignments calculation.
+                        extConfCh.changeAssignments(ByteUtils.toBytes(AffinityUtils.calculateAssignments(
+                                        baselineMgr.nodes(),
+                                        ch.partitions(),
+                                        ch.replicas())))
+                                // Table schema preparation.
+                                .changeSchemas(schemasCh -> schemasCh.create(
+                                        String.valueOf(INITIAL_SCHEMA_VERSION),
+                                        schemaCh -> {
+                                            SchemaDescriptor schemaDesc;
+
+                                            //TODO IGNITE-15747 Remove try-catch and force configuration
+                                            // validation here to ensure a valid configuration passed to
+                                            // prepareSchemaDescriptor() method.
+                                            try {
+                                                schemaDesc = SchemaUtils.prepareSchemaDescriptor(
+                                                        ((ExtendedTableView) ch).schemas().size(),
+                                                        ch);
+                                            } catch (IllegalArgumentException ex) {
+                                                throw new ConfigurationValidationException(ex.getMessage());
+                                            }
 
-                        if (ex instanceof TableAlreadyExistsException) {
-                            tblFut.completeExceptionally(ex);
-                        } else {
-                            LOG.error(IgniteStringFormatter.format("Table wasn't created [name={}]", name), ex);
+                                            schemaCh.changeSchema(SchemaSerializerImpl.INSTANCE.serialize(schemaDesc));
+                                        }
+                                ));
+                    });
+                }).exceptionally(t -> {
+                    Throwable ex = getRootCause(t);
 
-                            tblFut.completeExceptionally(ex);
-                        }
+                    if (ex instanceof TableAlreadyExistsException) {
+                        tblFut.completeExceptionally(ex);
                     } else {
-                        tblFut.complete(tables.get(name));
+                        LOG.error(IgniteStringFormatter.format("Table wasn't created [name={}]", name), ex);
+
+                        tblFut.completeExceptionally(ex);
```

Comment:
`tblFut` should also be removed from `tableCreateFuts`.

Reply:
You right, fixed.